src/base/dof_map_constraints.C

```diff
@@ -702,7 +702,7 @@ private:
               for (unsigned int qp=0; qp<n_qp; qp++)
                 {
                   // solution at the quadrature point
-                  OutputNumber fineval = 0;
+                  OutputNumber fineval(0);
```

Comment:
Are you guys both using the same compiler?  Which compiler?  Is it calling different constructors in the two cases?